crates/libcontainer/src/workload/mod.rs

```diff
@@ -60,6 +71,16 @@ pub trait Executor: CloneBoxExecutor {
     /// namespace and cgroups, and pivot_root into the rootfs. But this step
     /// runs before waiting for the container start signal.
     fn validate(&self, spec: &Spec) -> Result<(), ExecutorValidationError>;
+
+    /// Set environment variables for the container process to be executed.
```

Comment:
It'd be better to explain the situation this method is called. For example, this method is called the init process and hasn't cleared the host's envs yet.

Reply:
I've added a detailed description for the `setup_envs` in b20b8fa.